packages/modals/src/styled/StyledHeader.ts

```diff
@@ -22,21 +22,36 @@ export interface IStyledHeaderProps {
   isDanger?: boolean;
 }
 
+const sizeStyles = (props: ThemeProps<DefaultTheme>) => {
+  const verticalPadding = `${props.theme.space.base * 5}px`;
+  const startPadding = `${props.theme.space.base * 10}px`;
+  const endPadding = `${props.theme.space.base * 15}px`;
```

Comment:
The end padding should accommodate the entirety of the close button, which is 40px wide. Edge padding (40) + button padding (40) = 80px.

Reply:
> Is it reasonable to think that every single modal will contain a Close button? If not, this applies a rather severe padding imbalance to the header that can't be rationalized in the absence of a Close.

Oh snap. Didn't think of that. With this being positioned relative to the modal, is there even a way to do this accurately (without some JS scenanigans)?

Reply:
Feed it through context? Alternatively, it seems like the description use-case should be applying custom styling.

Reply:
Dunno if it helps but from a UX perspective, I don't think we'd recommend implementing modals without the Close button. Ideally there would be a non-janky fix for the severe padding if that scenario comes up but will leave that up to y'all